src/doc/reference.md

```diff
@@ -1413,6 +1413,24 @@ a = Animal::Cat { name: "Spotty".to_string(), weight: 2.7 };
 In this example, `Cat` is a _struct-like enum variant_,
 whereas `Dog` is simply called an enum variant.
 
+Enums have a discriminant. You can assign them explicitly:
+
+```
+enum Foo {
+    Bar = 123,
+}
+```
+
+If a discriminant isn't assigned, they start at zero, and add one for each
+variant, in order.
+
+You can cast an enum to get this value:
```

Comment:
One last nit: you can only do this if the enum is C-like, i.e. none of the variants have any data attached. For example, you can't do `None as u32`.